query/src/storages/storage_factory.rs

```diff
@@ -42,33 +42,58 @@ where
     }
 }
 
+#[derive(Default, Clone)]
+pub struct StorageDescriptor {
+    pub engine_name: String,
+    pub comment: String,
+}
+
 #[derive(Default)]
 pub struct StorageFactory {
     creators: RwLock<HashMap<String, Arc<dyn StorageCreator>>>,
+    descriptors: Vec<StorageDescriptor>,
```

Comment:
Thanks for the descriptor idea.
I think `StorageDescriptor` should be a part of table engine, it explicitly tell what this engine is, as defined by the engine's module not the `Factory`.

Reply:
what if every table engine implementes StorageCreateor trait and StorageCreator add interface desc() to return StorageDescriptor?

Reply:
> what if every table engine implementes StorageCreateor trait and StorageCreator add interface desc() to return StorageDescriptor?

Agreed.
But inclined to change `StorageCreateor` to `StorageDescriptor`, it's not only `create` matter also `comment`.